src/integration-test/java/com/commercetools/sync/integration/sdk2/services/impl/ShoppingListServiceImplIT.java

```diff
@@ -0,0 +1,304 @@
+import static com.commercetools.sync.integration.sdk2.commons.utils.ProductTypeITUtils.createProductType;
+import static com.commercetools.sync.integration.sdk2.commons.utils.ShoppingListITUtils.deleteShoppingListTestData;
+import static com.commercetools.sync.integration.sdk2.commons.utils.ShoppingListITUtils.deleteShoppingLists;
+import static com.commercetools.sync.integration.sdk2.commons.utils.TestClientUtils.CTP_TARGET_CLIENT;
+import static com.commercetools.sync.sdk2.products.ProductSyncMockUtils.PRODUCT_TYPE_RESOURCE_PATH;
+import static java.util.Collections.*;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.*;
+
+import com.commercetools.api.client.ProjectApiRoot;
+import com.commercetools.api.models.product.ProductDraft;
+import com.commercetools.api.models.product.ProductDraftBuilder;
+import com.commercetools.api.models.product.ProductVariantDraftBuilder;
+import com.commercetools.api.models.product_type.ProductType;
+import com.commercetools.api.models.product_type.ProductTypeResourceIdentifierBuilder;
+import com.commercetools.api.models.shopping_list.ShoppingList;
+import com.commercetools.api.models.shopping_list.ShoppingListChangeNameAction;
+import com.commercetools.api.models.shopping_list.ShoppingListChangeNameActionBuilder;
+import com.commercetools.api.models.shopping_list.ShoppingListDraft;
+import com.commercetools.api.models.shopping_list.ShoppingListDraftBuilder;
+import com.commercetools.api.models.shopping_list.ShoppingListLineItemDraft;
+import com.commercetools.api.models.shopping_list.ShoppingListLineItemDraftBuilder;
+import com.commercetools.api.models.shopping_list.TextLineItemDraft;
+import com.commercetools.api.models.shopping_list.TextLineItemDraftBuilder;
+import com.commercetools.sync.integration.sdk2.commons.utils.ShoppingListITUtils;
+import com.commercetools.sync.sdk2.services.ShoppingListService;
+import com.commercetools.sync.sdk2.services.impl.ShoppingListServiceImpl;
+import com.commercetools.sync.sdk2.shoppinglists.ShoppingListSyncOptions;
+import com.commercetools.sync.sdk2.shoppinglists.ShoppingListSyncOptionsBuilder;
+import io.vrap.rmf.base.client.ApiHttpResponse;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class ShoppingListServiceImplIT {
+  private ShoppingListService shoppingListService;
+
+  private ShoppingList shoppingList;
+  private List<String> errorCallBackMessages;
+  private List<Throwable> errorCallBackExceptions;
+
+  /**
+   * Deletes shopping list and products from the target CTP projects, then it populates the project
+   * with test data.
+   */
+  @BeforeEach
+  void setup() {
+    deleteShoppingListTestData(CTP_TARGET_CLIENT);
```

Comment:
I tried to comment out this deletion method. However, all the tests started failing. So for now I would keep it as it would require bigger changes to make this work and this is not part of the migration.